registry/consul/registry.go

```diff
@@ -104,22 +111,58 @@ func (r *consulRegistry) unregister(url common.URL) error {
 	return r.client.Agent().ServiceDeregister(buildId(url))
 }
 
-func (r *consulRegistry) Subscribe(url common.URL) (registry.Listener, error) {
+func (r *consulRegistry) subscribe(url *common.URL) (registry.Listener, error) {
 	var (
 		listener registry.Listener
 		err      error
 	)
 
 	role, _ := strconv.Atoi(r.URL.GetParam(constant.ROLE_KEY, ""))
 	if role == common.CONSUMER {
-		listener, err = r.getListener(url)
+		listener, err = r.getListener(*url)
 		if err != nil {
 			return nil, err
 		}
 	}
 	return listener, nil
 }
 
+//subscibe from registry
+func (r *consulRegistry) Subscribe(url *common.URL, notifyListener registry.NotifyListener) {
+	for {
+		if !r.IsAvailable() {
+			logger.Warnf("event listener game over.")
+			time.Sleep(time.Duration(RegistryConnDelay) * time.Second)
+			return
+		}
+
+		listener, err := r.subscribe(url)
+		if err != nil {
+			if !r.IsAvailable() {
```

Comment:
Why we need to check registry alive here?

Reply:
If registry is not alive, we do not need to resubsribe but only return the for loop.

Reply:
@hxmhlt Ok.